pkg/reconciler/broker/trigger/controller.go

```diff
@@ -69,7 +69,21 @@ func NewController(
 	r.sourceTracker = duck.NewListableTrackerFromTracker(ctx, source.Get, impl.Tracker)
 	r.uriResolver = resolver.NewURIResolverFromTracker(ctx, impl.Tracker)
 
-	triggerInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
+	triggerInformer.Informer().AddEventHandler(controller.HandleAll(
+		func(obj interface{}) {
+			if trigger, ok := obj.(*eventingv1.Trigger); ok {
+				broker, err := brokerInformer.Lister().Brokers(trigger.Namespace).Get(trigger.Spec.Broker)
+				if err != nil {
+					logger.Errorw("Failed to lookup Broker for Trigger", zap.Error(err))
+					return
+				}
+				label := broker.ObjectMeta.Annotations[brokerreconciler.ClassAnnotationKey]
+				if label == eventing.MTChannelBrokerClassValue {
+					impl.Enqueue(obj)
+				}
```

Comment:
We need the same filter logic on leader promotion, like this one https://github.com/knative-sandbox/eventing-kafka-broker/blob/c3ba96be7bd4de07c2d66c98f4a699578e798e3e/control-plane/pkg/reconciler/trigger/controller.go#L86

Reply:
Great catch!